### PR TITLE
Replace remaining occurences of 'typename std::...<>::type'

### PR DESCRIPTION
**Pull request by masterleinad**
Based on top of #4958, related to #4990.
Again, most of the work was done by something like
```bash
 git ls-files | xargs -n 1 perl -0777 -i -pe 's/typename std::([a-z]*_[a-z][a-z]+)(<((?>[^<>]+)|(?2))*\>)::type/std::\1_t\2/g'
```
plus some more manual changes.

### CODE REVIEW

File: core/src/impl/Kokkos_SimpleTaskScheduler.hpp [hunk 1]
```diff
@@ -221,7 +220,7 @@ class SimpleTaskScheduler
     // SharedAllocationRecord pattern
     using record_type =
         Impl::SharedAllocationRecord<memory_space,
-                                     Impl::DefaultDestroy<task_queue_type> >;
+                                     Impl::DefaultDestroy<task_queue_type>>;
```
Comment (ldh4) [inline core/src/impl/Kokkos_SimpleTaskScheduler.hpp:223]: I find it a bit interesting that this wasn't caught by clang-format check before.

**Review by ldh4 — APPROVED** [review]

**Review by cz4rs — APPROVED** [review]

**Comment by masterleinad** [comment]
> Please rebase to re-trigger the CI

Done.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by dalg24** [comment]
Failure was SYCL perftest.  Merging as is.

**Pull request merged**